CMSgov/mint-app
# Easi 2623/operational needs and solutions to user table

# EASI-2623 ,  EASI-2624

## Changes and Description

- Migrate Operational Needs to the User Table
- Migrate Operational Solutions to the User Table
- Create a createdByRelation and modifiedByRelation embedded struct
    - Can use this to house the shared logic of getting account information for these fields
    -  Currently, I just used them in the scope of the models for this ticket, but I think future work would be to embed these into base struct, and any other place that manually implements these fields ( like we do for user_account_relation) 

- Migrate Possible Operational Need to User Table
- Migrate Possible Operational Solution to User Table
- Migrate Possible Need Solution Link to User Table

Note, for configuration table changes, the migration was simpler because we know they all are only created by the system account. You will note that those migrations set those fields directly



## How to test this change
1. Bring up the application
2. Make a model plan (which will make operational needs)
3. Create a custom operational need
4. Create custom solution for that need
5. Ensure that you can save all the data


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
